javascript/webhooks-nodejs/createWebhook.js

```diff
@@ -0,0 +1,102 @@
+const axios = require("axios");
+const fs = require("fs");
+const path = require("path");
+require("dotenv").config();
+
+// File path for the .env file
+const envFilePath = path.join(__dirname, ".env");
+
+// Helper function to update the X-Hook-Secret in the .env file
+function updateXHookSecret(newSecret) {
+  let envContent = fs.readFileSync(envFilePath, "utf8");
+  envContent = envContent.replace(
+    /X_HOOK_SECRET=.*/,
+    `X_HOOK_SECRET=${newSecret}`
+  );
+  fs.writeFileSync(envFilePath, envContent);
+  console.log(`The X-Hook-Secret stored in .env is: ${newSecret}`);
+}
+
+// Helper function to read X-Hook-Secret from the .env file
+function getXHookSecret() {
+  const envContent = fs.readFileSync(envFilePath, "utf8");
+  const match = envContent.match(/X_HOOK_SECRET=(.*)/);
+  return match ? match[1] : "";
+}
+
+// Async function to delete the webhook
+// Documentation: https://developers.asana.com/reference/deletewebhook
+async function deleteWebhook(webhookId, personalAccessToken) {
+  const url = `https://app.asana.com/api/1.0/webhooks/${webhookId}`;
+
+  try {
+    await axios.delete(url, {
+      headers: {
+        Authorization: `Bearer ${personalAccessToken}`,
+      },
+    });
+    console.log(`Webhook ${webhookId} deleted due to mismatch.`);
+  } catch (error) {
+    console.error(
+      "Error deleting webhook:",
+      error.response ? error.response.data : error.message
+    );
+  }
+}
+
+// Async function to create the webhook
+// Documentation: https://developers.asana.com/reference/createwebhook
+async function createWebhook(targetUri, objectId, filter, resourceType) {
+  const personalAccessToken = process.env.PAT;
+  const url = "https://app.asana.com/api/1.0/webhooks";
+
+  try {
+    const response = await axios.post(
+      url,
+      {
+        data: {
+          resource: objectId,
+          target: targetUri,
+          filters: [{ action: filter, resource_type: resourceType }],
+        },
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${personalAccessToken}`,
+        },
+      }
+    );
+
+    const xHookSecret = response.data["X-Hook-Secret"];
+    const storedSecret = getXHookSecret();
+
+    // Compare the secrets
+    // Read more about validation during the webhook handshake: https://developers.asana.com/docs/webhooks-guide#the-webhook-handshake
+    if (xHookSecret !== storedSecret) {
```

Comment:
Hmm not sure if I can trigger this case. I might need you to show me how to hit this case. I tried changing my `.env` file's `X-Hook-Secret` before running and I don't encounter this case. Might be my terminal or something

Reply:
Hmm I'm unable to re-produce the use case above. I created a random helper function to programmatically set `X_HOOK_SECRET` in the .env file to `abc`:

```js
function setXHookSecretToAbc() {
  const newSecret = "abc";
  let envContent = fs.readFileSync(envFilePath, "utf8");
  envContent = envContent.replace(
    /X_HOOK_SECRET=.*/,
    `X_HOOK_SECRET=${newSecret}`
  );
  fs.writeFileSync(envFilePath, envContent);
  console.log(`The X-Hook-Secret in .env has been updated to: ${newSecret}.`);
}
```

Then I called it just before I compared the stored secret vs the one from Asana's `201`:

```js
const xHookSecret = response.data["X-Hook-Secret"];

console.log(`The original X_HOOK_SECRET from .env before the re-write is ${getXHookSecret()}`)
setXHookSecretToAbc();
console.log(
  `After the re-writen, the X_HOOK_SECRET from .env re-write is now ${getXHookSecret()}`
);

const storedSecret = getXHookSecret();
```

The console output was:

```bash
The original X_HOOK_SECRET from .env before the re-write is 11111111111111111111111111111111
The X-Hook-Secret in .env has been updated to: abc.
After the re-writen, the X_HOOK_SECRET from .env re-write is now abc
X-Hook-Secrets do not match! Potential security issue. Deleting webhook with GID 2222222222222222.
Webhook 2222222222222222 deleted due to mismatch.
```

Reply:
This case will be harder to test since it requires pretending to be Asana and sending back a 201 response with a fake `X-Hook-Secret`.  Adding a comment, here explaining why we do this check should help developers understand the purpose.

It looks like you've already added a comment so this should be good